### ISSUE TITLE
TimberPost constructor doing bad guesswork

### ISSUE THREAD

**Issue by teohhanhui** [jarednova/timber#520]
Problem introduced by 3d633c60873b778e4c12906b617d91eff4ef9ad7

We cannot assume that `$wp_query->queried_object_id` does not refer to a taxonomy ID or anything else... Which means this is totally broken when used in the loop on a category archive page, for example.

I don't even think it was a correct fix for #344 as doing `new TimberPost()` in the loop now gets you the page (how counter-intuitive!). The correct behavior would be for TimberPost to always have the same result as get_post

**Comment by jarednova** [comment]
Thanks for raising this @teohhanhui -- I've created a PR here to resolve:

https://github.com/jarednova/timber/pull/521

In regards to #344, I did some additional testing and confirmed the fix. Even though it's an edge-case, I'd like to preserve this functionality. With regards to "doing a `new TimberPost();` in the loop" -- part of my philosophy behind Timber is to avoid the assumed context being "in the loop". I really hate the idea that the same code performs differently in different places without different params being clearly sent.

... but anyway, I'd really appreciate your feedback on #521 to see if it resolves the problems you're having. Thanks!
